pkg/ruler/evaluator_remote_test.go

```diff
@@ -411,6 +414,77 @@ func TestRemoteEvalUnsupportedResultResponse(t *testing.T) {
 	require.ErrorContains(t, err, fmt.Sprintf("unsupported result type: %q", loghttp.ResultTypeStream))
 }
 
+// TestRemoteEvalQueryTags validates that proper query tags are passed
+func TestRemoteEvalQueryTags(t *testing.T) {
+	defaultLimits := defaultLimitsTestConfig()
+	limits, err := validation.NewOverrides(defaultLimits, nil)
+	require.NoError(t, err)
+
+	now := time.Now()
+
+	cli := mockClient{
+		handleFn: func(ctx context.Context, in *httpgrpc.HTTPRequest, opts ...grpc.CallOption) (*httpgrpc.HTTPResponse, error) {
+			var queryTags string
+			for _, h := range in.Headers {
+				if h.Key == string(httpreq.QueryTagsHTTPHeader) {
+					queryTags = strings.Join(h.Values, ",")
+				}
+			}
+			require.Equal(t, "ruler,Source=ruler,Rule=foo,Kind=alerting", queryTags)
```

Comment:
I think the more correct/readable approach here would be to assign this value to a variable and move the `require.Equal` out into the main scope of the test.

For example, if this function is never called the test will pass.